app/routes/users/components/PhotoConsents.js

```diff
@@ -0,0 +1,182 @@
+      </h5>
+      <div className={styles.statusContainer}>{getConsentStatus()}</div>
+      <div>
+        <ConfirmModalWithParent
+          closeOnConfirm={true}
+          title={`Trekke bildesamtykke på ${presentableDomain}`}
+          message={`Er du sikker på at du vil trekke bildesamtykket ditt for ${toReadableSemester(
+            consent
+          )} på ${presentableDomain}? Dersom du ønsker å fjerne noen spesifikke bilder, kan du i stedet sende en mail til pr@abakus.no med informasjon om hvilke bilder du vil fjerne.`}
+          onConfirm={() => updateConsent({ ...consent, isConsenting: false })}
+        >
+          <Button
+            disabled={!isMe}
+            className={
+              consent.isConsenting === false ? styles.selectedConsentBtn : ''
+            }
+          >
+            Nei
+          </Button>
+        </ConfirmModalWithParent>
+        <Button
+          disabled={!isMe}
+          onClick={() => updateConsent({ ...consent, isConsenting: true })}
+          className={
+            consent.isConsenting
+              ? cx(styles.consentBtn, styles.selectedConsentBtn)
+              : styles.consentBtn
+          }
+        >
+          Ja
+        </Button>
+      </div>
+    </>
+  );
+};
+
+const PhotoConsents = ({
+  photoConsents,
+  username,
+  updatePhotoConsent,
+  userId,
+  isMe,
+}: {
+  photoConsents: Array<PhotoConsent>,
+  username: string,
+  updatePhotoConsent: (
+    PhotoConsent: PhotoConsent,
+    username: string,
+    userId: Number
+  ) => Promise<*>,
+  userId: Number,
+  isMe: boolean,
+}) => {
+  const semesterLabels = [
+    ...new Set(
+      photoConsents
+        .slice(0)
+        .sort((a, b) => {
+          if (a.year === b.year) {
+            return a.semester === 'spring' ? 1 : -1;
+          }
+
+          return a.year > b.year ? -1 : 1;
+        })
+        .map((photoConsent) => toReadableSemester(photoConsent))
+    ),
+  ];
+
+  const getConsentFromSemesterLabel = (
+    domain: PhotoConsentDomain,
+    semesterLabel: string
+  ): ?PhotoConsent => {
+    const match = /(?<semester>\S+) (?<year>\d+)/.exec(semesterLabel);
+    const year = Number(match?.groups?.year);
+    const semester = match?.groups?.semester === 'våren' ? 'spring' : 'autumn';
+    return photoConsents.find(
+      (pc) =>
+        pc.domain === domain && pc.year === year && pc.semester === semester
+    );
+  };
+
+  const [selectedSemester, setSelectedSemester] = useState(semesterLabels[0]);
+
+  const updateConsent = (consent: PhotoConsent) =>
+    updatePhotoConsent(consent, username, userId);
+  return (
+    <Flex column={true}>
+      <label htmlFor="select-semester">
+        <h3>Semester</h3>
+      </label>
+      <Select
+        name="select-semester"
+        clearable={false}
+        options={semesterLabels.map((semester) => ({
+          value: semester,
+          label: semester,
+        }))}
+        value={{
+          value: selectedSemester,
```

Comment:
Why is this not just an object representing the semester?? And you could get rid of the `getConsentFromSemesterLabel`

Reply:
Yeah, it was kinda weird. Fixed. 👍🏻